src/dkg_bipolynomial.erl

```diff
@@ -27,9 +36,23 @@ add(PolyA, PolyB) ->
 sub(PolyA, PolyB) ->
     merge(PolyA, PolyB, fun erlang_pbc:element_sub/2).
 
+is_zero(Poly) ->
+    tuple_size(Poly) == 0.
+
 degree(Poly) ->
     tuple_size(Poly) - 1.
 
+cmp(PolyA, PolyB) ->
+    %% checks f(x, y) - g(x, y) = 0
+    %% subtracting a polynomial from itself should yield all 0s
+    %% and all the coefficients should match
+    lists:all(fun(X) ->
+                      X
+              end,
+              [ erlang_pbc:element_cmp(lookup([I, J], PolyA), lookup([I, J], PolyB)) || I <- lists:seq(1, degree(PolyA)), J <- lists:seq(1, degree(PolyB))])
+    andalso
+    is_zero(dkg_bipolynomial:sub(PolyA, PolyB)).
```

Comment:
Why do both a compare and subtract? You should also check they both have the same degree or the list comprehension will fail.

Reply:
Agreed on checking the degree first, the need for zero test is the same as for poly test, basically f(x, y) = g(x, y) if f(x, y) - g(x, y) = h(x, y); then h(x, y) is 0 for all values of x and y. Atlteast theoretically.